src/redux/navigation.js

```diff
@@ -9,13 +10,20 @@ module.exports.navigationReducer = (state, action) => {
         state = '';
     }
     switch (action.type) {
+    case Types.SET_MODE:
+        return action.mode;
```

Comment:
This reducer is poorly named, but is responsible for the search term (as seen below, `SET_SEARCH_TERM` sets the state to the action's `searchTerm`. This means that if you set the state to `mode`, you're setting the search term to `mode`. This prevents the Search page from making the correct API request for search results.

Since the search mode is confined to the search page, I don't think you need to worry about Redux. You can remove the changes you have in this file.